main.cpp

```diff
@@ -13,75 +13,571 @@ What you create in this project will be used as the basis of Project 5 in the co
+    */
+};
+
+
+/*
+Thing 1) Coffee shop
+5 properties:
+    1) the number of baristas (int)
+    2) the number of coffee machines (int)
+    3) the number of registers (int)
+    4) the number of tables (int)
+    5) the types of coffee (std::string)
+3 things it can do:
+    1) make a coffee
+    2) grind coffee beans
+    3) modify customer names
+ */
+
+struct CoffeeShop
+{
+    // 1) the number of baristas (int)
+    int numBaristas = 2;
+    // 2) the number of coffee machines (int)
+    int numCoffeeMachines = 4;
+    // 3) the number of registers (int)
+    int numRegisters = 2;
+    // 4) the number of tables (int)
+    int numTables = 12;
+    // 5) the types of coffee (std::string)
+    std::string brewMethod = "pour over";
+
+    struct Coffee
+    {
+        std::string type = "columbian";
+        std::string roast = "dark";
+    };
+
+    struct Customer
+    {
+        std::string name = "Customer 1";
+        float customerPhoneNumber = 15555555555.0f; //replit suggested 'long' be used, but that's not in the lesson
+        float customerID = 1.0f; // I'll also float this then
+        bool rewardsMember = false;
+        float rewardsBalance = 0.0f;
+
+        bool useRewardsPoints(float rewardsPoints); //returns if there wasn't enough balance
+        void newMemberPromotion(float incentiveAmount);
+        bool contactCustomer(std::string msg = "Your order is ready.");
+    };
+
+    // 1) make a coffee
+    bool brewCoffee(Coffee coffeeType, int size, std::string brewType, bool cream, bool sugar, std::string customerName); //returns "coffee ready" - I think you'd do this for 'urgent' asynchronous tasks like this?
+    // 2) grind coffee beans
+    bool grindCoffee(Coffee coffeeType, int courseness, Customer customerA);
+    // 3) modify customer names
+    void renameCustomer(Customer customerA);
+
+    Coffee standardBrew;
+};
+
+/*
+Thing 2) Invoice application
+5 properties:
+    1) the number of invoices (int)
+    2) the number of clients (int)
+    3) the outstanding balance (float)
+    4) the number of templates (int)
+    5) the types of jobs (std::string)
+3 things it can do:
+    1) create an invoice
+    2) check if an invoice is late
+    3) calculate outstanding balance
+ */
+
+struct InvoiceManager
+{
+    // 1) the number of invoices (int)
+    int numInvoices = 234;
+    // 2) the number of clients (int)
+    int numClients = 32;
+    // 3) the outstanding balance (float)
+    float outBalance = 10345.24f;
+    // 4) the number of templates (int)
+    int numTemplates = 4;
+    // 5) the type of job (std::string)
+    std::string workType = "mastering";
+
+    struct Invoice
+    {
+        std::string clientName;
+        int invoiceNumber;
+        std::string workType = "post";
+        float workTime = 0.0f;
+        float totalBalance = 0.0f;
+        float dueDate;
+        bool overdue = false;
+
+        void download(std::string format = "pdf");
+        void paid();
+        void duplicate();
```

Comment:
`paid` is a state of something.  
it is not an action, which is what functions are supposed to be.
pick a better name
- [ ] revise